14_contains/solution/contains-solution.spec.js

```diff
@@ -0,0 +1,50 @@
+const contains = require('./contains-solution');
+
+describe('contains', () => {
+  const object = {
+    data: {
+      duplicate: 'e',
+      info: {
+          duplicate: 'e',
+          magicNumber: 44,
+          stuff: {
+              thing: {
+                banana: NaN,
+                moreStuff: {
+                    something: 'foo',
+                    answer: [42]
+                }
+              }
+          }
+      }
+    }
+  }
```

Comment:
I think a good edge case to add here would be a `null` value somewhere in the object. The solution you have doesn't handle this, so it'd also require some tinkering with that.

Reply:
Great catch

Reply:
`typeof null === 'object'` javascript is a beautiful language :smirk:

Reply:
Yeah you always have to be careful checking `typeof === null`, this is what happens when you make a language in just 10 days xD